kernel/src/history_manager/mod.rs

```diff
@@ -44,14 +49,13 @@ type Timestamp = i64;
 /// Once created, the [`LogHistoryManager`] does not automatically update with newer versions
 /// of the table. All timestamp queries are limited to the state captured in the [`Snapshot`]
 /// provided during construction.
-#[allow(unused)]
 #[derive(Debug)]
-pub(crate) struct LogHistoryManager {
+pub struct LogHistoryManager {
     log_segment: LogSegment,
     snapshot: Arc<Snapshot>,
+    commit_to_timestamp_cache: RefCell<HashMap<Url, Timestamp>>,
```

Comment:
does a cache belong in kernel?

Reply:
Yeah I wondered about that as well. I decided to go with caching because the cache is c*O(size of log segment). And the constant c is rly small.